ts/sagas/wallet.ts

```diff
@@ -347,6 +363,18 @@ function* showWalletOrSelectPsp(idWallet: number, paymentId?: string) {
 function* continueWithPaymentMethodsHandler(
   _: PaymentRequestContinueWithPaymentMethods
 ) {
+  /**
+   * ask and check PIN before procedd with payment
+   */
+  // Retrieve the configured PIN from the keychain
+  const storedPin: SagaCallReturnType<typeof getPin> = yield call(getPin);
```

Comment:
instead of fetch the PIN from the state, pass it as a parameter to the saga, this saga is likely forked from the `startup` saga, that validates the PIN and can provide a PIN that is always defined, thus avoiding the "impossible" state of undefined PIN